src/components/split-button/split-button-rtl.test.tsx

```diff
@@ -0,0 +1,912 @@
+import React from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent, { UserEvent } from "@testing-library/user-event";
+import StyledButton from "components/button/button.style";
+import SplitButton from "./split-button.component";
+import Button from "../button";
+import { SizeOptions } from "../button/button.component";
+import {
+  testStyledSystemMargin,
+  toCSSCase,
+} from "../../__spec_helper__/test-utils";
+import I18nProvider from "../i18n-provider";
+
+jest.mock("../../__internal__/utils/helpers/guid", () => () => "guid-12345");
+
+const sizes = ["small", "medium", "large"] as const;
+type SizeType = typeof sizes[number];
+
+const singleButton = (size: SizeType = "medium") => (
+  <Button size={size} key="testKey">
+    Single Button
+  </Button>
+);
+
+const multipleButtons = [
+  <Button key="testKey1">Extra Button 1</Button>,
+  <Button key="testKey2">Extra Button 2</Button>,
+  <Button key="testKey3">Extra Button 3</Button>,
+];
```

Comment:
My personal opinion on these is there used repeatedly and they're so simple that I don't think anyone is going to struggle to understand seeing them used as one is a single child the other is three etc. IMO this is the right level of DRYness we should go for, I don't think things like this hurt readability and are a helpful shortcut rather than bloating the code by defining it repeatedly